src/main/scala/gitbucket/core/controller/api/ApiIssueCommentControllerBase.scala

```diff
@@ -65,9 +65,39 @@ trait ApiIssueCommentControllerBase extends ControllerBase {
   })
 
   /*
-   * v. Edit a comment
-   * https://developer.github.com/v3/issues/comments/#edit-a-comment
+   * v. Update an issue comment
+   * https://developer.github.com/v3/issues/comments/#update-an-issue-comment
    */
+  patch("/api/v3/repos/:owner/:repository/issues/comments/:id")(readableUsersOnly { repository =>
+    val commentId = params("id")
```

Comment:
There's a lot of nesting going on, do you think it would be a better idea to simplify this by doing the `isEditable` check in the `updateCommentByApi` function?

I think the reason you do it here is because you want to preserve the `Unauthorized` error right? Otherwise you only have the possibility of returning `Option[(Issue, Int)]` in `updateCommentByApi`. Maybe you can try using the `Either[A, B]` type here and make `updateCommentByApi` return `Either[ActionResult, Option[String]]`?

Alternatively, maybe you can do the following
```Scala
patch("/api/v3/repos/:owner/:repository/issues/comments/:id")(readableUsersOnly { repository =>
    val commentId = params("id")
    (for {
      issueComment <- getComment(repository.owner, repository.name, commentId)
      issue <- getIssue(repository.owner, repository.name, issueComment.issueId.toString)
    } yield {
      if (isEditable(repository.owner, repository.name, issue.openedUserName)) {
        val maybeBody = extractFromJsonBody[CreateAComment].map(_.body)
        val maybeUpdatedBody = updateCommentWithPermission(repository, issue, issueComment, maybeBody)
        Right(maybeUpdatedBody)
      } else {
        Left(Unauthorized())
      }
    }).map {
        case Right(value) => value.getOrElse(NotFound())
        case Left(value)  => value
      }
      .getOrElse(NotFound())
  })

  private[this] def updateCommentWithPermission(
    repoInfo: RepositoryInfo,
    issue: Issue,
    issueComment: IssueComment,
    maybeBody: Option[String]
  ): Option[String] =
    for {
      _ <- updateCommentByApi(repoInfo, issue, issueComment.commentId.toString, maybeBody)
      issueComment <- getComment(repoInfo.owner, repoInfo.name, issueComment.commentId.toString)
    } yield {
      JsonFormat(
        ApiComment(
          issueComment,
          RepositoryName(repoInfo),
          issue.issueId,
          ApiUser(context.loginAccount.get),
          issue.isPullRequest
        )
      )
    }
```

I'm definitely not a Scala expert, just wanted to get you thoughts on this approach 🙂

Reply:
Any thoughts @onukura ?

Reply:
@jyoo980 Thanks for your proposal! Yeah. now my code has so many nests... your way looks simple and nice. I am not scala expert too so I'm not sure which is better... 

@SIkebe @takezoe What do you think about this?

Reply:
@onukura your way is no better than my way in terms of what it achieves 🙂

The best (and worst) thing about code is that there's so many ways to do the same thing, and it's not always clear which way is the best. The work you've done is _awesome_!!!

Reply:
In this case, the combination of for comprehension and map seems better to reduce nests, but it's just my preference. For sure, as @jyoo980 mentioned, there are many ways to do the same thing in Scala and it's always tough to say which way is better while this flexibility is one beneficial points of Scala.

Reply:
Also, in this case, using `for` comprehension (or the combination of `flatMap` and `map`) and translate `None` to `NotFound` at last might reduce pattern matching as follows:
```scala
val result = getCommnet(...).flatMap { issueComment =>
  getIssue(...).map { issue =>
    ..
  }
}

result match {
  case Some(response) => response
  case None => NotFound()
}
```